hudi-client/hudi-java-client/src/test/java/org/apache/hudi/common/table/read/HoodieFileGroupReaderOnJavaTestBase.java

```diff
@@ -113,4 +132,65 @@ public void commitToTable(List<HoodieRecord> recordList, String operation, boole
       }
     }
   }
+
+  @Override
+  public void commitSchemaToTable(InternalSchema schema, Map<String, String> writeConfigs, String historySchemaStr) {
+    String tableName = writeConfigs.get(HoodieTableConfig.HOODIE_TABLE_NAME_KEY);
+    Schema avroSchema = AvroInternalSchemaConverter.convert(schema, getAvroRecordQualifiedName(tableName));
+
+    StorageConfiguration<?> storageConf = getStorageConf();
+    String basePath = getBasePath();
+
+    Map<String, String> finalWriteConfigs = new HashMap<>(writeConfigs);
+    finalWriteConfigs.put(HoodieCleanConfig.AUTO_CLEAN.key(), "false");
+    finalWriteConfigs.put(HoodieCleanConfig.FAILED_WRITES_CLEANER_POLICY.key(),
+        HoodieFailedWritesCleaningPolicy.NEVER.name());
+    finalWriteConfigs.put(HoodieArchivalConfig.AUTO_ARCHIVE.key(), "false");
+
+    HoodieJavaClientTestHarness.TestJavaTaskContextSupplier taskContextSupplier = new HoodieJavaClientTestHarness.TestJavaTaskContextSupplier();
+    HoodieJavaEngineContext context = new HoodieJavaEngineContext(getStorageConf(), taskContextSupplier);
+
+    HoodieWriteConfig config = HoodieWriteConfig.newBuilder()
+        .withPath(basePath)
+        .withSchema(avroSchema.toString())
+        .withEngineType(EngineType.JAVA)
+        .withProps(finalWriteConfigs)
+        .build();
+
+    try (HoodieJavaWriteClient<?> client = new HoodieJavaWriteClient<>(context, config)) {
+
+      HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder()
+          .setConf(storageConf)
+          .setBasePath(basePath)
+          .setTimeGeneratorConfig(config.getTimeGeneratorConfig())
+          .build();
+
+      WriteOperationType operationType = WriteOperationType.ALTER_SCHEMA;
+      String commitActionType = CommitUtils.getCommitActionType(operationType, metaClient.getTableType());
+
+      String instantTime = client.startCommit(commitActionType);
+      client.setOperationType(operationType);
+
+      HoodieActiveTimeline timeline = metaClient.getActiveTimeline();
+      TimelineLayout layout = metaClient.getTimelineLayout();
+      HoodieInstant requested = layout.getInstantGenerator()
+          .createNewInstant(HoodieInstant.State.REQUESTED, commitActionType, instantTime);
+
+      HoodieCommitMetadata metadata = new HoodieCommitMetadata();
+      metadata.setOperationType(operationType);
+
+      timeline.transitionRequestedToInflight(requested, Option.of(metadata));
+
+      Map<String, String> extraMeta = new HashMap<>();
+      long schemaId = Long.parseLong(instantTime);
+      InternalSchema withId = schema.setSchemaId(schemaId);
+      extraMeta.put(SerDeHelper.LATEST_SCHEMA, SerDeHelper.toJson(withId));
```

Comment:
This is not a big deal here but generally it's good to get in the habit of using `Collections.singletonMap` when making maps with a single element. It helps cut down on the overhead of the map. This ends up being a big win when dealing with anything happening many times like a per-row operation so it is good to get in the habit of doing this.